src/csharp/Microsoft.Spark/Sql/Types/Date.cs

```diff
@@ -0,0 +1,94 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.Spark.Sql.Types
+{
+    /// <summary>
+    /// Represents Date containing year, month, and day.
+    /// </summary>
+    public class Date
+    {
+        /// <summary>
+        /// Constructor for Date class.
+        /// </summary>
+        /// <param name="dt">DateTime object</param>
+        public Date(DateTime dt)
+        {
+            Year = dt.Year;
+            Month = dt.Month;
+            Day = dt.Day;
+        }
+
+        /// <summary>
+        /// Constructor for Date class.
+        /// </summary>
+        /// <param name="year">The year (1 through 9999)</param>
+        /// <param name="month">The month (1 through 12)</param>
+        /// <param name="day">The day (1 through the number of days in month)</param>
+        public Date(int year, int month, int day)
+        {
+            Year = year;
+            Month = month;
+            Day = day;
+        }
+
+        /// <summary>
+        /// Returns the year component of the date.
+        /// </summary>
+        public int Year { get; private set; }
+
+        /// <summary>
+        /// Returns the month component of the date.
+        /// </summary>
+        public int Month { get; private set; }
+
+        /// <summary>
+        /// Returns the day component of the date.
+        /// </summary>
+        public int Day { get; private set; }
+
+        /// <summary>
+        /// Readable string representation for this type.
+        /// </summary>
+        public override string ToString() => $"{Month}/{Day}/{Year}";
+
+        /// <summary>
+        /// Readable string representation for this type using the specified format.
+        /// </summary>
+        /// <param name="format">A standard or custom date and time format string</param>
+        public string ToString(string format) => new DateTime(Year, Month, Day).ToString(format);
+
+        /// <summary>
+        /// Checks if the given object is same as the current object.
+        /// </summary>
+        /// <param name="obj">Other object to compare against</param>
+        /// <returns>True if the other object is equal.</returns>
+        public override bool Equals(object obj) =>
+            ReferenceEquals(this, obj) ||
+            ((obj is Date date) && Year.Equals(date.Year) &&
+            Month.Equals(date.Month) && Day.Equals(date.Day));
+
+        /// <summary>
+        /// Returns the hash code of the current object.
+        /// </summary>
+        /// <returns>The hash code of the current object</returns>
+        public override int GetHashCode() => base.GetHashCode();
+
+        /// <summary>
+        /// Returns DateTime object describing this type.
```

Comment:
Can we have <returns> tag since this is public API?

Reply:
Tag is added. Thanks.